xcm/xcm-builder/src/currency_adapter.rs

```diff
@@ -170,4 +170,23 @@ impl<
 			.map_err(|e| XcmError::FailedToTransactAsset(e.into()))?;
 		Ok(what.clone().into())
 	}
+
+	fn transfer_asset(
+		asset: &MultiAsset,
+		from: &MultiLocation,
+		to: &MultiLocation,
+	) -> result::Result<Assets, XcmError> {
+		log::trace!(target: "xcm::currency_adapter", "transfer_asset asset: {:?}, from: {:?}, to: {:?}", asset, from, to);
+		let amount: u128 =
+			Matcher::matches_fungible(asset).ok_or(Error::AssetNotFound)?.saturated_into();
```

Comment:
I might suggest we do a `try_into` rather than saturated into here since failing to convert a value is probably better than saturating here.

Reply:
Huh interesting, that means I don't even need this intermediate `amount` variable at all, since it looks like the only purpose that it's there is to attempt and convert it to a `u128` and saturate it, before it gets converted back to `Balance` as seen in `balance_amount`.

Reply:
Is there some other code which is making you think this might be the better way to go?

Do you generally agree with my assessment that an error here is nicer than some saturating behavior?

Reply:
Yeah, the code in `deposit_asset` and `withdraw_asset` both use `saturated_into`, and if we are going to change here, then it makes sense to change it on both of those functions.

I don't really have an opinion on how it should work, but I'm assuming that there's a reason why the existing code chooses to saturate into a `u128` first and converting it back to a `Balance`.

Reply:
I think you're right @KiChjang --- there's really no reason to convert from `Currency::Balance` to `u128` and then back again. It should just be:

```
let balance_amount = Matcher::matches_fungible(asset).ok_or(Error::AssetNotFound)?;
```